### DISCUSSION TITLE
Velox Release Process

### DISCUSSION THREAD

**Discussion by majetideepak** [facebookincubator/velox#9694] [Ideas]
Current Status
- Velox API changes are less frequent.
- Velox modules are still coupled. eg: Connector and DWIO, etc.
- Prestissimo: "Advance Velox" are less frequent.
- Prestissimo uses a Velox submodule. No version number is available.

The Case for a Velox Release
- Helps clients develop custom extensions as each release provides a stable API.
- Allows implementing features such as dynamic plugins. Eg. Load functions, and connectors at runtime.
- Forces us to decouple modules. Dynamic library support can help reduce binary size.
- Follow release cycles of other Meta projects such as folly, proxygen, etc... I.e, every Monday.

Improving the build is a prerequisite for the release process. We need dynamic library support.
The discussion here is tracking build improvements: https://github.com/facebookincubator/velox/discussions/8657

**Comment by kgpai** [comment]
@majetideepak  Should we tackle versioning also as part of the release process ?

**Reply by majetideepak** [reply]
Yes, the version number is the key part of each release.
